makefiles/tools/avrdude.inc.mk

```diff
@@ -6,10 +6,16 @@ DEBUGSERVER_FLAGS = "-g -j usb :$(DEBUGSERVER_PORT)"
 DEBUGGER_FLAGS = "-x $(RIOTBOARD)/$(BOARD)/dist/gdb.conf $(ELFFILE)"
 DEBUGGER = $(DIST_PATH)/debug.sh $(DEBUGSERVER_FLAGS) $(DIST_PATH) $(DEBUGSERVER_PORT)
 
-# make the flasher port configurable (e.g. with atmelice the port is usb)
-# defaults to terminal's serial port if not configured
-AVRDUDE_PORT    ?= $(PORT)
-PROGRAMMER_FLAGS = -P $(AVRDUDE_PORT) $(FFLAGS_EXTRA)
+PROGRAMMER_FLAGS = -p $(CPU:atmega=m)
```

Comment:
on MacOS this doesn't seem to work, for Arduino-Mega2560 I get:

```
avrdude -c stk500v2 -p atmega2560 -P ...
```

while on master it is `m2560` bc explicitly set. HOWEVER, funnily it even works with that - but maybe that's just a coincidence and we should not rely on that.

This does not seem to be related to `make`, I tried with the default on MacOS which is version 3.81 and with one installed via HomeBrew which is 4.2.1.

Reply:
The full name (e.g. `atmega2560` instead of `m2560`) was actually used by the jiminy board as default. In the doc I only found the short names. But if the full name is supported officially, I would actually prefer that. I'll look that up.

If not, I think using `subst` to replace atmega by m should work anywhere.

Reply:
OK, the official doc does not contain any hint that the full name is supported. I consider this as an "Easter Egg" and I would favor to not rely on that.